corporate/lib/stripe.py

```diff
@@ -1496,18 +1496,42 @@ def configure_fixed_price_plan(self, fixed_price: int, sent_invoice_id: str | No
             current_plan.save(update_fields=["status", "next_invoice_date"])
             return f"Fixed price {required_plan_tier_name} plan scheduled to start on {current_plan.end_date.date()}."
 
+        # TODO: Use normal 'pay by invoice' flow for fixed-price plan offers,
+        # which requires handling automated license management for these plan
+        # offers via that flow.
         if sent_invoice_id is not None:
             sent_invoice_id = sent_invoice_id.strip()
-            # Verify 'sent_invoice_id' before storing in database.
+            # Verify 'sent_invoice_id' and 'stripe_customer_id' before
+            # storing in database.
             try:
                 invoice = stripe.Invoice.retrieve(sent_invoice_id)
                 if invoice.status != "open":
                     raise SupportRequestError(
                         "Invoice status should be open. Please verify sent_invoice_id."
                     )
+                invoice_customer_id = invoice.customer
+                if not invoice_customer_id:  # nocoverage
+                    raise SupportRequestError(
+                        "Invoice missing Stripe customer ID. Please review invoice."
+                    )
+                if customer.stripe_customer_id and customer.stripe_customer_id != str(
+                    invoice_customer_id
+                ):  # nocoverage
+                    raise SupportRequestError(
+                        "Invoice Stripe customer ID does not match. Please attach invoice to correct customer in Stripe."
```

Comment:
This seemed better than overwriting a stripe_customer_id that is attached to the Customer object in our database. I thought about sending the `customer.stripe_customer_id` in the support message, but again I don't think this will come up too much in practice.

Reply:
Probably better to have it, but yes I expect 99% of the time we can just look it up by email in the Stripe panel anyway. (or actually I guess it's already visible on that support page).